### PR TITLE
Add IMessageGenerator interface

### PR DESCRIPTION
**Pull request by matt0bi**
**What this PR does / why we need it**:

This PR adds a `IMessageGenerator` interface, which aims to define the behavior of objects that create messages passed between plugins.

**Which issue(s) this PR fixes**

N/A

**Special notes for your reviewer**:

None.

**Release note**:

```release-note
NONE
```

@spotify/alf

### CODE REVIEW

File: gordon_janitor/interfaces.py [hunk 1]
```diff
@@ -117,3 +117,24 @@ class IPublisher(IGenericPlugin):
...
         """Initialize a Publisher Plugin client."""
+
+
+class IMessageGenerator(Interface):
+    """Create messages that can be sent from plugin to plugin."""
+    async def generate_msg(self, data_item):
+        """Create a message based off a data item.
+
+        Args:
+            data_item (any): source data.
```
Comment (econchick) [inline gordon_janitor/interfaces.py:128]: Are we sure we don't want to enforce types?

**Review by econchick — COMMENTED** [review]
I realize I have yet to do this with the other interfaces, but I think it's a little confusing at what this is used for - do you mind writing some prose docs for this?

File: gordon_janitor/interfaces.py [hunk 2]
```diff
@@ -117,3 +117,24 @@ class IPublisher(IGenericPlugin):
     """
     def __init__(config, changes_channel, metrics=None):
         """Initialize a Publisher Plugin client."""
+
+
+class IMessageGenerator(Interface):
+    """Create messages that can be sent from plugin to plugin."""
+    async def generate_msg(self, data_item):
```
Comment (econchick) [inline gordon_janitor/interfaces.py:124]: Do these need to be async?

**Comment by matt0bi** [comment]
After reading through the comments on https://github.com/spotify/gordon-janitor-gcp/pull/12 and our recent talks, I'm closing this out. Thanks for reviewing this @econchick !

**Pull request closed without merging**